cli/flox-rust-sdk/src/models/lockfile.rs

```diff
@@ -159,47 +258,178 @@ impl LockedManifestCatalog {
     /// Convert a locked manifest to a list of installed packages for a given system
     /// in a format shared with the pkgdb based locked manifest.
     pub fn list_packages(&self, system: &System) -> Vec<InstalledPackage> {
-        self.groups
+        self.packages
             .iter()
-            .filter(|group| &group.system == system)
-            .flat_map(|group| {
-                group.page.packages.iter().cloned().map(|package| {
-                    let priority = self
-                        .manifest
-                        .install
-                        .values()
-                        .find(|install| {
-                            install.pkg_path == package.attr_path
-                                && install
-                                    .package_group
-                                    .as_deref()
-                                    .unwrap_or(DEFAULT_GROUP_NAME)
-                                    == group.name
-                        })
-                        .and_then(|install| install.priority);
-
-                    InstalledPackage {
-                        name: package.name,
-                        rel_path: package.attr_path,
-                        info: PackageInfo {
-                            description: Some(package.description),
-                            broken: package.broken,
-                            license: Some(package.license),
-                            pname: package.pname,
-                            unfree: package.unfree,
-                            version: Some(package.version),
-                        },
-                        priority,
-                    }
-                })
+            .filter(|package| &package.system == system)
+            .map(|package| {
+                let priority = self
+                    .manifest
+                    .install
+                    .iter()
+                    .find(|(install_id, _)| install_id == &&package.name)
+                    .and_then(|(_, descriptor)| descriptor.priority);
+
+                let package = package.clone();
+
+                InstalledPackage {
+                    name: package.name,
+                    rel_path: package.attr_path,
+                    info: PackageInfo {
+                        description: Some(package.description),
+                        broken: package.broken,
+                        license: Some(package.license),
+                        pname: package.pname,
+                        unfree: package.unfree,
+                        version: Some(package.version),
+                    },
+                    priority,
+                }
             })
             .collect()
     }
+
+    /// Produce a lockfile for a given manifest using the catalog service.
+    ///
+    /// If a seed lockfile is provided, packages that are already locked
+    /// will constrain the resolution.
+    pub async fn new(
```

Comment:
nit
```suggestion
    pub async fn lock_manifest_with_catalog_client(
```

Reply:
Yeah youre right `new` seems more like a pure cheap'ish operation.
lets call it `lock_manifest` for now (its already in the _Catalog_ context via the struct name).